### ISSUE TITLE
Dropdown scroll doesn't work in modal

### ISSUE THREAD

**Issue by dgandhi1993** [themesberg/flowbite-svelte#653]
### Describe the bug

While adding a static or dynamic dropdown to an open modal, the scroll of the dropdown doesn't seem to work at all.

### Reproduction

Try adding a scrolling dropdown to a modal to see the reproduction. 
Here is an [example](https://stackblitz.com/edit/sveltejs-kit-template-default-s2t2wh?file=src/routes/+page.svelte).

### Flowbite version and System Info

```shell
npmPackages:
    @sveltejs/kit: 1.0.1 
    flowbite-svelte: 0.29.13 
    svelte: 3.55.0 
    vite: 4.0.2
```

**Comment by dgandhi1993** [comment]
Can someone help here? Been stuck with this issue for quite some time now.

**Comment by jjagielka** [comment]
I can't reproduce your issue. Check here: [REPL](https://svelte.dev/repl/ad52b5a0f4254347a20a4616512e060d?version=3.59.1)

I've noticed your example uses quite an old version of `flowbite-svelte`. Can you please upgrade and check again?

**Comment by shinokada** [comment]
Thank you for your contribution! We are closing the issue for now due to inactivity. If you encounter any other problems in the future, please feel free to open a new issue. We appreciate your help in making our project better!

**Comment by agrear** [comment]
I'm on `flowbite-svelte` v1.31.0 inside `tauri` v2.9.5.

Using the slightly modified code from the (no longer working) REPL:
```svelte
<script>
  import { Button, Modal, Dropdown, DropdownItem } from 'flowbite-svelte';

  let scrollingModal = $state(false);
  const items = Array(18).fill(0).map((_, i) => i * i);
</script>

<Button color="red" onclick={() => scrollingModal = true}>Scrolling modal</Button>

<Modal id="scrolling-modal" title="Terms of Service" bind:open={scrollingModal}>
  <p class="text-base leading-relaxed text-gray-500 dark:text-gray-400">
    With less than a month to go before the European Union enacts new consumer privacy laws for its citizens, companies around the world are updating their terms of service agreements to comply.
  </p>
  <Button color="green">Dropdown button</Button>
  <Dropdown class="h-40 overflow-y-auto">
    {#each items as item}
      <DropdownItem>Dashboard {item}</DropdownItem>
    {/each}
  </Dropdown>
</Modal>
```

I encounter two issues:
1. Opening the modal also opens the dropdown.
2. Clicking *anywhere* including the dropdown itself, closes it. This makes it impossible to scroll the items without using the mouse wheel.

---

Edit: Just saw that 1. is addressed in #1910.
